transport_nantes/topicblog/views.py

```diff
@@ -1375,3 +1383,34 @@ def click_received_handler(sender, mail_obj, click_obj, *args, **kwargs):
     logger.info(
         f"\nclick_received signal received for message {aws_message_id}\n"
         f"SendRecord class : {send_record_class} ID : {send_record_id}")
+
+
+def mark_moribund_and_delete(slug_queryset):
+    """Delete or prepare to delete old objects.
+
+    Mark for deletion objects that are moribund.
+    Delete objects that have been marked moribund for long enough.
+
+    I really don't like that the queryset language doesn't permit me
+    to use a function, as this means that the functions defined on the
+    model (which can only apply to individual instances) can't be used
+    here, and the queryset definitions of moribund and deletable can't
+    be used there, even if I made a new manager.
+
+    Somewhat worse, I don't see how to do the SQL queries efficiently
+    in django's query language in order to call update().  So, rather
+    inefficiently, I'm loading all unpublished objects for the slug.
+    In principle, that isn't too inefficient, since after a few weeks
+    there likely won't be any.
+
+    """
+    slug_queryset.filter(publication_date=None)
+    for tb_object in slug_queryset:
+        if tb_object.is_deletable():
+            logger.info(f"Deleting object {tb_object.id}.")
+            tb_object.delete()
+        elif tb_object.is_moribund() and \
+             tb_object.scheduled_for_deletion_date is None:
+            logger.info(f"Marking as moribund object {tb_object.id}.")
+            tb_object.scheduled_for_deletion_date = datetime.now(timezone.utc)
+            tb_object.save()
```

Comment:
When saving, tb_object.date_modified will be set to today because of auto_now (https://docs.djangoproject.com/fr/4.0/ref/models/fields/#django.db.models.DateField.auto_now ). Doing so this line in models, `is_moribund` : 

```python
        if self.first_publication_date is None and \
           ((datetime.now(timezone.utc) - self.date_modified).days >=
                self.K_MORIBUND_DELAY_DAYS):
```
Will evaluate to today - today >= 15 -> 0 >=15 and return False, while it should be saying yes, it's moribund.
The consequence of this is that you don't get to see the countdown in the listone.

This doesn't show up in your tests because you manually set the date_modified attribute.

Reply:
Wait, does it reset on every save??  😢

Reply:
Right.  Done.